### PR TITLE
Garland fix and updates

### PR DESCRIPTION
**Pull request by ElderJoy**
- fix issue with division by zero in first scene setup
- fix web UI
- update scene default settings
- update palettes
- got rid of redundant variables in main class
- add animations:
  -  random run animation
  - anim_crossing
- update animations:
  - anim_glow
  - anim_spread

### CODE REVIEW

File: code/espurna/garland/animations/anim_dolphins.h [hunk 1]
```diff
@@ -51,25 +51,22 @@ class AnimDolphins : public Anim {
...
+        int len = secureRandom(10, 30);
         int speed = secureRandom(1, 3);
-        int dir = 1;
+        int dir = randDir();
         int head = 0;
         int start;
         Color color;
         std::unique_ptr<Color[]> points;
-        Dolphin(Palette* pal, uint16_t numLeds) : start(secureRandom(0, numLeds - len)), color(pal->getRndInterpColor()) {
+        Dolphin(Palette* pal, uint16_t numLeds) : numLeds(numLeds), start(head ? secureRandom(0, numLeds - len) : secureRandom(len, numLeds)), color(pal->getRndInterpColor()) {
```
Comment (mcspr) [inline code/espurna/garland/animations/anim_dolphins.h]: s/head/dir/?
(unless I am missing how it is anything but 0 from init above)

File: code/espurna/garland/animations/color_wave.h [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        Palette *_palette,
+        uint16_t _waveLen,
+        bool _cleanColors,
+        byte _fade = 0,
+        Color* _pixelCache = nullptr,
+        WaveType _type = WaveType::Wave,
+        float _speed = secureRandom(5, 15) / 10.0,
+        int _dir = randDir(),
+        bool _startEmpty = false)
+        : numLeds(_numLeds), palette(_palette), waveLen(_waveLen), cleanColors(_cleanColors), fade(_fade), pixelCache(_pixelCache), type(_type), speed(_speed), dir(_dir) {
```
Comment (mcspr) [inline code/espurna/garland/animations/color_wave.h]: (maybe ot)

just a reminder, named fields are a thing even in gcc4.8 (but only if one doesn't add default values)
Keeps ctors arglist from growing, imo keeps it more readable.
```cpp
struct Params {
  int a;
  int b;
};

...

class Bar {
  Bar(Params);
private:
  Params _params;
};

...

Bar(Params{ .a = 1, .b = 2 });
```

File: code/espurna/garland/animations/color_wave.h [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    Color getLedColor(uint16_t ledNum) {
+        uint16_t real_led_num = dir > 0 ? ledNum : numLeds - ledNum - 1;
+
+        if (real_led_num > head) {
+            return Color();
+        }
+
+        if (pixelCache && !(pixelCache[real_led_num] == Color())) {
```
Comment (mcspr) [inline code/espurna/garland/animations/color_wave.h]: color_obj.empty()?

File: code/espurna/garland/animations/anim_dolphins.h (hunk 1 above)
Comment (ElderJoy) [reply code/espurna/garland/animations/anim_dolphins.h]: Yes, it should be dir. Thanks! Fixed.

File: code/espurna/garland/animations/color_wave.h (hunk 2 above)
Comment (ElderJoy) [reply code/espurna/garland/animations/color_wave.h]: Agree, the code is more clear now.
By-the-way, Rust has more sophisticated constructor possibilities for default params.

File: code/espurna/garland/animations/color_wave.h (hunk 3 above)
Comment (ElderJoy) [reply code/espurna/garland/animations/color_wave.h]: Yes. Also update to is_empty() and add static empty constructor for Color.

**Comment by mcspr** [comment]
Thanks!

**Pull request merged**